lib/index.js

```diff
@@ -345,23 +345,32 @@ exports.init = function (sequelize, optionsArg) {
       opt.transaction = instance.context.transaction;
     }
 
-    if (options.enableCompression) {
-      var previousVersion = {};
-      var currentVersion = {};
+    let previousVersion = {};
+    let currentVersion = {};
 
+    if (options.enableCompression) {
       _.forEach(opt.defaultFields, function (a) {
         if (!instance._previousDataValues[a].dataValues && !instance.dataValues[a].dataValues) {
           previousVersion[a] = instance._previousDataValues[a];
           currentVersion[a] = instance.dataValues[a];
         }
       });
     } else {
-      var previousVersion = instance._previousDataValues;
-      var currentVersion = instance.dataValues;
+      previousVersion = instance._previousDataValues;
+      currentVersion = instance.dataValues;
     }
-    // Supported nested models.
-    previousVersion = _.omitBy(previousVersion, function (i) {return typeof i === 'object'});
-    currentVersion = _.omitBy(currentVersion, function (i) {return typeof i === 'object'});
+    
+    const isSequelizeModel = (value) => {
+      // Models associated by `hasMany` will appear as arrays of Models, we need to strip those out too.
+      if (_.isArray(value)) {
+        return value[0] instanceof sequelize.Model;
+      }
+      return value instanceof sequelize.Model
+    }
+
+    // Prevents diffing associated models leading to circular calls
+    previousVersion = _.omitBy(previousVersion, val => isSequelizeModel(val));
+    currentVersion = _.omitBy(currentVersion, val => isSequelizeModel(val));
```

Comment:
So, if I read this correctly, this will filter out anything that is a Sequelize Model?

Reply:
you got it sparky